Upgrade to h5py==3.0.0 causes Keras-related tests to fail

[The most recent cron build](https://travis-ci.org/github/neuropoly/spinalcordtoolbox/builds/740371045) failed across all OS jobs. The following tests are failing: 

```
FAILED unit_testing/test_deepseg_gm.py::TestCore::test_segment_volume 
FAILED unit_testing/test_deepseg_lesion.py::test_segment 
FAILED unit_testing/test_deepseg_sc.py::test_deep_segmentation_spinalcord[params0] 
FAILED unit_testing/test_deepseg_sc.py::test_deep_segmentation_spinalcord[params1]
FAILED unit_testing/test_qc_parallel.py::test_many_qc 
FAILED unit_testing/test_sct_compute_mtsat.py::test_with_json_sidecar[input_params0]  
FAILED unit_testing/test_sct_compute_mtsat.py::test_with_json_sidecar[input_params1]
FAILED unit_testing/test_straightening.py::test_straighten  
FAILED unit_testing/cli/test_cli_sct_analyze_lesion.py::test_sct_analyze_lesion_backwards_compat[subprocess] 
FAILED unit_testing/cli/test_cli_sct_deepseg_gm.py::test_sct_deepseg_gm_backwards_compat[subprocess] 
FAILED unit_testing/cli/test_cli_sct_deepseg_lesion.py::test_sct_deepseg_lesion_backwards_compat[subprocess] 
FAILED unit_testing/cli/test_cli_sct_deepseg_sc.py::test_sct_deepseg_sc_backwards_compat[subprocess] 
```

[The cron build from the night previous](https://travis-ci.org/github/neuropoly/spinalcordtoolbox/builds/740117624) passes, so I imagine this is an upstream issue. I'm looking into this immediately, and will comment + update title when I have a better idea.

[Diff between logs](https://www.diffchecker.com/AhicwDJT)

```diff

Check Python executable.............................[OK]
  Using bundled python 3.6.12 |Anaconda, Inc.| (default, Sep  8 2020, 23:10:56) 
[GCC 7.3.0] at /home/travis/build/neuropoly/spinalcordtoolbox/python/envs/venv_sct/bin/python
Check if data are installed.........................[OK]
Check if colored is installed.......................[OK] (1.4.2)
Check if dipy is installed..........................[OK] (1.2.0)
Check if futures is installed.......................[OK]
- Check if h5py is installed..........................[OK] (2.10.0)
+ Check if h5py is installed..........................[OK] (3.0.0)
Check if ivadomed (2.0.2) is installed..............[OK] (2.0.2)
Check if Keras (2.1.5) is installed.................[OK] (2.1.5)
Check if matplotlib is installed....................[OK] (3.3.2)
Check if nibabel is installed.......................[OK] (3.2.0)
Check if numpy is installed.........................[OK] (1.19.3)
- Check if pandas is installed........................[OK] (1.1.3)
+ Check if pandas is installed........................[OK] (1.1.4)
Check if psutil is installed........................[OK] (5.7.3)
Check if pyqt5 (5.11.3) is installed................[OK] (5.11.3)
Check if pytest is installed........................[OK] (6.1.2)
Check if pytest-cov is installed....................[OK] (__version__ = '2.10.1')
Check if raven is installed.........................[OK]
Check if requests is installed......................[OK] (2.24.0)
Check if requirements-parser is installed...........[OK] (0.2.0)
Check if scipy is installed.........................[OK] (1.5.3)
Check if scikit-image is installed..................[OK] (0.17.2)
Check if scikit-learn is installed..................[OK] (0.23.2)
Check if tensorflow (1.5.0) is installed............[OK] (1.5.0)
Check if torch (1.5.0+cpu) is installed.............[OK] (1.5.0+cpu)
Check if torchvision (0.6.0+cpu) is installed.......[OK] (0.6.0+cpu)
Check if xlwt is installed..........................[OK] (1.3.0)
Check if tqdm is installed..........................[OK] (4.51.0)
Check if transforms3d is installed..................[OK] (0.3.1)
Check if urllib3 is installed.......................[OK] (1.25.11)
Check if pytest_console_scripts is installed........[OK]
Check if pytest-xdist is installed..................[OK] (2.1.0)
Check if spinalcordtoolbox is installed.............[OK]
Check ANTs compatibility with OS ...................[OK]
Check PropSeg compatibility with OS ................[OK]
Check if DISPLAY variable is set....................[FAIL]
```

Based on (some) of the error messages, I'm willing to bet the update to `h5py` is the issue here:

```python
f = <Closed HDF5 file>
layers = [<keras.engine.topology.InputLayer object at 0x7f0a5c569cf8>, <keras.layers.convolutional.Conv2D object at 0x7f0a5c569...tional.Conv2D object at 0x7f0a5c529d68>, <keras.layers.normalization.BatchNormalization object at 0x7f0a5c57ae10>, ...]
reshape = False
    def load_weights_from_hdf5_group(f, layers, reshape=False):
        """Implements topological (order-based) weight loading.
    
        # Arguments
            f: A pointer to a HDF5 group.
            layers: a list of target layers.
            reshape: Reshape weights to fit the layer when the correct number
                of values are present but the shape does not match.
    
        # Raises
            ValueError: in case of mismatch between provided layers
                and weights file.
        """
        if 'keras_version' in f.attrs:
>           original_keras_version = f.attrs['keras_version'].decode('utf8')
E           AttributeError: 'str' object has no attribute 'decode'
```

The remaining errors are `FileNotFoundError`s for images in our testing data. But, I'll start looking into the `h5py` update for now to see if it's the sole cause or if there's something else.

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,7 +1,7 @@
 colored
 dipy
 futures
-h5py
+h5py==2.10.0
 ivadomed==2.0.2
 Keras==2.1.5
 matplotlib

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,7 +1,9 @@
 colored
 dipy
 futures
-h5py
+# h5py is pinned to minor than 3 due to issues with Keras/TF
+# https://github.com/tensorflow/tensorflow/issues/44467
+h5py<3.0
 ivadomed==2.0.2
 Keras==2.1.5
 matplotlib

--- a/requirements.txt
+++ b/requirements.txt
@@ -3,7 +3,7 @@ dipy
 futures
 # h5py is pinned to minor than 3 due to issues with Keras/TF
 # https://github.com/tensorflow/tensorflow/issues/44467
-h5py<3.0
+h5py~=2.10.0
 ivadomed==2.0.2
 Keras==2.1.5
 matplotlib

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,7 +1,7 @@
 colored
 dipy
 futures
-h5py==2.10.0
+h5py
 ivadomed==2.0.2
 Keras==2.1.5
 matplotlib

